internal/manifests/targetallocator/annotations.go

```diff
@@ -26,15 +26,14 @@ import (
 const configMapHashAnnotationKey = "opentelemetry-targetallocator-config/hash"
 
 // Annotations returns the annotations for the TargetAllocator Pod.
-func Annotations(instance v1alpha1.OpenTelemetryCollector) map[string]string {
+func Annotations(instance v1alpha1.OpenTelemetryCollector, configMap *v1.ConfigMap) map[string]string {
```

Comment:
This was one of the more significant changes, rather than creating the config map in this function its passed down and optionally nil. cc @swiatekm-sumo who may have thoughts here.

Reply:
I'm fine with this, it's basically just moving the evaluation a level up.

Reply:
Agreed